core/src/main/java/com/linecorp/armeria/common/logging/MappedContentPreviewerFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2015 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.logging;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Supplier;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.RequestContext;
+
+final class MappedContentPreviewerFactory implements ContentPreviewerFactory {
+    final List<Entry<MediaType, Supplier<ContentPreviewer>>> entries;
+
+    MappedContentPreviewerFactory(Map<MediaType, Supplier<ContentPreviewer>> map) {
+        this(requireNonNull(map, "map").entrySet());
+    }
+
+    MappedContentPreviewerFactory(Iterable<Entry<MediaType, Supplier<ContentPreviewer>>> entries) {
+        this.entries = ImmutableList.copyOf(requireNonNull(entries, "entries"));
```

Comment:
It might be an overkill but how about doing null checks on `Entry.key()` and `Entry.value()`?